services/docsrs/docsrs.tester.js

```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Passing docs')
+  .get('/tokio/0.3.0.json')
+  .expectBadge({ label: 'docs@0.3.0', message: 'passing' })
+
+t.create('Failing docs')
+  .get('/tensorflow/0.16.1.json')
+  .expectBadge({ label: 'docs@0.16.1', message: 'failing' })
```

Comment:
For these service tests (which can roughly be viewed as high level e2e tests/integration tests with upstream APIs), we want to be sure that we exercise the different routes/inputs for our badge, and that the resultant badge is of the correct "shape"

So here, we'd want to have a test that does not provide an explicit version to provide verification that the default/latest badges are operating correctly.

We also typically use Joi schemas as validators to ensure that the message is _an_ expected value (from a set of known/possible values) as opposed to a literal specific value, especially since in most cases that literal value will be dynamic as things change upstream. In this particular case, I think it's fine to keep these literal tests in here since the doc build states for those specific crate versions is immutable, and we also have to manually directly map an the upstream response value (boolean build_status) to the corresponding message whereas in most other badges we're using the standard helpers (`renderBuildStatusBadge`, `isBuildStatus`, etc.)

However, for the additional test that needs to be added to cover the latest/no-specific-version route, you'll want to use Joi so that the test won't be brittle:

```js
messge: Joi.allow('passing', 'failing'),
```